### ISSUE TITLE
[HIPIFY][MSVS][clang][bug] Hipification started failing on Windows with MSVS 17.4.0

### ISSUE THREAD

**Issue by emankov** [ROCm-Developer-Tools/HIPIFY#693]
**[Synopsis]**

The in-process tool `clang-tool` has started to be invoked with the `unknown` triple:

```cmd
"clang-tool" "-cc1" "-triple" "unknown" "-target-sdk-version=11.8" "-aux-triple" "nvptx-nvidia-cuda" "-fsyntax-only" ... "-x" "cuda" ...
```
**[Expected behaviour]**
Should be the MSVC triple like `x86_64-pc-windows-msvcXX.YY.ZZZZZ`:
```cmd
"clang-tool" "-cc1" "-triple" "x86_64-pc-windows-msvc19.34.31933" "-target-sdk-version=11.8" "-aux-triple" "nvptx64-nvidia-cuda" "-fsyntax-only" ... "-x" "cuda" ...
```
**[Suggestion]**
It looks like Clang's Driver has not supported yet the new C++ compiler, shipped with MSVS 17.4.0

**Comment by emankov** [comment]
Hipification with the latest LLVM release 15.0.5 is ok for MS VS 2022 17.4.x. So, the problem with the unknown target triple is the LLVM 16.0.0git regression.

**[ToDo]**
1. Triage the affection LLVM change(s)
2. File an issue to LLVM
3. Fix it in LLVM

**Comment by emankov** [comment]
Fixed with a workaround by #712
